mlir/tools/mlir-tblgen/EnumsGen.cpp

```diff
@@ -23,18 +25,32 @@
 #include "llvm/TableGen/Record.h"
 #include "llvm/TableGen/TableGenBackend.h"
 
+using namespace mlir;
+using namespace mlir::tblgen;
+
 using llvm::formatv;
 using llvm::isDigit;
 using llvm::PrintFatalError;
 using llvm::raw_ostream;
 using llvm::Record;
 using llvm::RecordKeeper;
-using llvm::StringRef;
-using mlir::tblgen::Attribute;
-using mlir::tblgen::EnumAttr;
-using mlir::tblgen::EnumAttrCase;
-using mlir::tblgen::FmtContext;
-using mlir::tblgen::tgfmt;
+
+/// Returns true if 'subNamespace' is a sub-namespace of 'parentNamespace'.
+/// I.e. 'subNamespace' is contained within 'parentNamespace'.
+static bool isSubNamespace(ArrayRef<StringRef> subNamespace,
+                           StringRef parentNamespace) {
+  SmallVector<StringRef> parentNamespaces;
+  llvm::SplitString(parentNamespace, parentNamespaces, "::");
```

Comment:
would be better to use a regex match here

Reply:
we have a regex option for splitting op definitions.

Reply:
I prototyped this real quick and I am not 100% sure how I feel about it. 
What I like about the current solution is that its simple yet useful, and intuative in the way that C++ namespaces work. Specifying something like `-enum-namespace=test` e.g. simply generates all namespaces that have `test::` in them. 
The equivalent regex to do so would be `^test(::.*)?$` (or just `test(::.*)?` on the CLI if we make the implementation wrap it in `^...$`). 

What do you think?